### DISCUSSION TITLE
Next/Image outputs remote images with incorrect query parameters

### DISCUSSION THREAD

**Discussion by xxczaki** [vercel/next.js#43492] [Help]
### What version of Next.js are you using?

10.2.0 and 10.2.1-canary.5

### What version of Node.js are you using?

15.4.0

### What browser are you using?

Brave 1.24.84 (Chromium: 90.0.4430.93)

### What operating system are you using?

macOS

### How are you deploying your application?

next start (Digital Ocean server)

### Describe the Bug

I want to load a remote image (e.g. https://files.mypolitics.pl/mypolitics2/cdnv2-myplogo_e3e80327aa.png) via next/image. I added the `files.mypolitics.pl` domain to next.config.js, so that it works with the `Image` component. Unfortunately when I start the dev server, it produces the following error:

<img width="592" alt="Zrzut ekranu 2021-05-12 o 22 03 39" src="https://user-images.githubusercontent.com/22645979/118038119-00237e80-b36f-11eb-8e64-ab876a2c2e58.png">

For some reason, Next.js incorrectly appends `&w=64&q=75` (instead of `?w=64&q=75`) to the URL. Note that this is not the case just for this image - all remote images work like that.

### Expected Behavior

I would expect Next.js to handle these images correctly.

### To Reproduce

Try loading the `https://files.mypolitics.pl/mypolitics2/cdnv2-myplogo_e3e80327aa.png` via next/image. I stumbled upon this issue during migration to next/image - the project is open-source, and the migration PR is available here: 

https://github.com/mypolitics/mypolitics-3/pull/134

**Comment by timneutkens** [comment]
Can you try opening the url, it should show an error that can be traced back to what is wrong.
This is not related to the querystring as the external url is provided as as part of the querystring, notice `?url=` before the encoded external url.

**Comment by xxczaki** [comment]
@timneutkens Thanks for a quick reply - I didn't notice the `?url=` at first. When I visit any of the URLs, the error is always: `The requested resource isn't a valid image.`. No further information is available in the console.

**Comment by icyJoseph** [comment]
~We are also having this issue when trying to upgrade. Formally, when using 10.0.x, PNG images with transparent backgrounds show with the typical black fill for the transparent areas. We thought of upgrading to see if this issue would resolve, but when upgrading we start to get 400 errors, similarly to this closed issue https://github.com/vercel/next.js/issues/23837, because our images have content type octet-stream.~

Any work around for PNG's? Other than optimization opt-out? Or would it just work to change them to `image/*`?

Btw, it'd be great to have a way to debug the next/image process. Print which url it is taking in, did it manage to get the image? Is it resizing it, which were the original dimensions versus the output? did it send the result to the client? How feasible is it to implement such a thing?

**Comment by jankaifer** [comment]
Since there is no bug here - I'll move this into a discussion.
We will keep an eye on this conversation.

**Comment by detrumi** [comment]
So in my understanding:
- `Next/Image` requests `_next/image?url=file.png&w=320`
- Next.js then requests the image using `example.org/file.png&w=320`

But what if the CDN doesn't support such query strings? Is our only option to opt out of image optimizations?

**Reply by gerardnico** [reply]
Url has nothing to do with cdn cache.

Check:

https://datacadamia.com/web/http/cache/intermediate

On how you can control them.

Cdn may have special http headers to add cache features.

**Reply by detrumi** [reply]
I didn't mention any cache, the issue is that the domain where our images live is managed by a third party, which disallows such query strings without a `?`.
